datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -370,6 +373,34 @@ func (f *ChainDataFetcher) handleRequestByType(reqType cfTypes.RequestType, shou
 		f.updateCheckpoint(ev.Block.Number().Int64())
 	}
 	handledBlockNumberGauge.Update(ev.Block.Number().Int64())
+	return nil
+}
+
+func (f *ChainDataFetcher) resetChainCh() {
+	for {
+		select {
+		case <-f.chainCh:
+		default:
+			return
+		}
+	}
+}
+
+func (f *ChainDataFetcher) resetRequestCh() {
+	for {
+		select {
+		case <-f.reqCh:
+		default:
+			return
+		}
+	}
+}
```

Comment:
Why don't you just assign a new channel to `f.chain` and `f.reqCh`?

Reply:
Because I want to reuse those channels when the fetching jobs rerun.